### ISSUE TITLE
MPSNDArray.mm:782: failed assertion; bufer is not large enough Mac M1 MPS

### ISSUE THREAD

**Issue by astrowonk** [pytorch/pytorch#86152]
### 🐛 Describe the bug

This appears a duplicate or a regression of #77886. I'm trying to use MPS with OpenAI's whisper. This had been failing but I saw that ` aten::index_put_impl ` was now supported so I tried again:

```{python}
import whisper
import torch
device = torch.device('mps')
model.transcribe("test dictation.m4a",language='english',verbose=True,fp16=False)


```

Which leads to a crash:
```
/usr/local/opt/miniforge3/lib/python3.9/site-packages/whisper/decoding.py:628: UserWarning: The operator 'aten::repeat_interleave.self_int' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/mps/MPSFallback.mm:11.)
  audio_features = audio_features.repeat_interleave(self.n_group, dim=0)
/AppleInternal/Library/BuildRoots/a0876c02-1788-11ed-b9c4-96898e02b808/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:782: failed assertion `[MPSNDArray, initWithBuffer:descriptor:] Error: buffer is not large enough. Must be 201452 bytes
```

I'm running `torch-1.13.0.dev20221003`

### Versions

```
Collecting environment information...
PyTorch version: 1.13.0.dev20221003
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.6 (arm64)
GCC version: Could not collect
Clang version: 14.0.0 (clang-1400.0.29.102)
CMake version: version 3.23.3
Libc version: N/A

Python version: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 17:00:33)  [Clang 13.0.1 ] (64-bit runtime)
Python platform: macOS-12.6-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.0
[pip3] torch==1.13.0.dev20221003
[pip3] torchaudio==0.13.0.dev20221003
[pip3] torchvision==0.14.0.dev20221003
[conda] numpy                     1.22.0           py39h61a45d2_0    conda-forge
[conda] torch                     1.13.0.dev20221003          pypi_0    pypi
[conda] torchaudio                0.13.0.dev20221003          pypi_0    pypi
[conda] torchvision               0.14.0.dev20221003          pypi_0    pypi
```

cc @kulinseth @albanD @malfet @DenisVieriu97 @razarmehr @abhudev

**Comment by malfet** [comment]
Ok, I have a reproducer using base model, let me have a look into what is going on.

**Comment by malfet** [comment]
Ok, looks like GELU on mps is broken for half types, but should be trivial to fix. Here is one line reproducer:
```
% python -c "import torch;print(torch.nn.GELU()(torch.rand(2, device='mps', dtype=torch.half)))"
self.sizes()=[2] dtypeHalf
loc("mps_multiply"("(mpsFileLoc): /AppleInternal/Library/BuildRoots/4883e71d-37bd-11ed-b0ef-b25c5e9b9057/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShadersGraph/mpsgraph/MetalPerformanceShadersGraph/Core/Files/MPSGraphUtilities.mm":228:0)): error: input types 'tensor<2xf16>' and 'tensor<1xf32>' are not broadcast compatible
LLVM ERROR: Failed to infer result type(s).
zsh: abort      python -c 
```

**Comment by malfet** [comment]
And here is a single line reproducer for 2nd problem:
```
% python -c "import torch;sz=2800;x=torch.rand(1, sz, device='mps');x[:,sz-128:]=-1"
```

**Comment by vashat** [comment]
Hi! Any progress on this issue?

**Comment by malfet** [comment]
Yeah, I have a PR to fix the remaining crash, but then it produces a bogus results, so I sort of abandoned the pursuit...

**Comment by mpottinger** [comment]
I have what seems to be the same kind of issue running HashNeRFpytorch

/AppleInternal/Library/BuildRoots/2d9b4df9-4b93-11ed-b0fc-2e32217d8374/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:794: failed assertion `[MPSNDArray, initWithBuffer:descriptor:] Error: buffer is not large enough. Must be 262144 bytes

**Comment by anubhavbhatti** [comment]
I am also facing the same issue:

/AppleInternal/Library/BuildRoots/5b8a32f9-5db2-11ed-8aeb-7ef33c48bc85/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:794: failed assertion `[MPSNDArray, initWithBuffer:descriptor:] Error: buffer is not large enough. Must be 576 bytes

Is there a fix?

**Comment by nilsec** [comment]
Same here
> /AppleInternal/Library/BuildRoots/20d6c351-ee94-11ec-bcaf-7247572f23b4/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:782: failed assertion `[MPSNDArray, initWithBuffer:descriptor:] Error: buffer is not large enough. Must be 292280 bytes

**Comment by keefmoon** [comment]
I also have the same issue, attempting to run OpenAI's Whisper with MPS support

> /AppleInternal/Library/BuildRoots/c651a45f-806e-11ed-a221-7ef33c48bc85/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:794: failed assertion `[MPSNDArray, initWithBuffer:descriptor:] Error: buffer is not large enough. Must be 1007280 bytes

**Comment by DenisVieriu97** [comment]
> I also have the same issue, attempting to run OpenAI's Whisper with MPS support
> 
> > /AppleInternal/Library/BuildRoots/c651a45f-806e-11ed-a221-7ef33c48bc85/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:794: failed assertion `[MPSNDArray, initWithBuffer:descriptor:] Error: buffer is not large enough. Must be 1007280 bytes

@keefmoon this should be fixed in the latest pytorch nightly. Could you please try that and let me know if you still see any issues?

**Comment by keefmoon** [comment]
Weirdly I pulled the nightly yesterday and had the issue, but I just pulled the nightly again, which seems to be the same version, but it's now it's running without issue. Thanks!

**Comment by keefmoon** [comment]
I agree that the Whisper now running on the GPU on the M1 Max in my Mac Studio is *way* slower than I expected. But I guess that isn't relevant to this specific crash.

**Comment by vashat** [comment]
> > It is over twice as slow as CPU only but it worked.
> 
> I agree that the Whisper now running on the GPU on the M1 Max in my Mac Studio is **way** slower than I expected. But I guess that isn't relevant to this specific crash.

Same here, about 30% slower using GPU on my M1.  Has anyone tried with the Ventura 13.3 beta that has support for int64 repeats mask?

**Comment by mukulpatnaik** [comment]
> Could not run 'aten::empty.memory_format' with arguments from the 'SparseMPS' backend. This could be because the operator doesn't exist for this backend, or was omitted during the selective/custom build process (if using custom build).

Getting the same issue

**Comment by astrowonk** [comment]
Consider upvoting #87886, which is the SparseMPS `aten::empty.memory_format` issue.

**Comment by kulinseth** [comment]
@skotapati please take a look at this issue.

**Comment by skotapati** [comment]
Does not repro on MacOS 14.6.1 with latest torch nightly, please reopen if the issue is observed again
